modules/core/src/main/scala/org/tessellation/domain/snapshot/programs/Download.scala

```diff
@@ -225,6 +220,19 @@ object Download {
               case Some(snapshot) =>
                 globalSnapshotContextFns
                   .createContext(context, lastSnapshot, snapshot)
+                  .flatTap(newContext =>
+                    snapshotStorage
+                      .hasCorrectSnapshotInfo(snapshot.ordinal, snapshot.stateProof)
+                      .flatMap(Applicative[F].unlessA(_) {
```

Comment:
```suggestion
                      .ifM(Applicative[F].unit, {
```
Is this equivalent?  If so, is it easier to read?

Reply:
hah, I did that before and changed that to applicative unlessA. I can change that with some other PR.